tool/tbot/tshwrap/wrap_test.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tshwrap
+
+import (
+	"encoding/json"
+	"path/filepath"
+	"testing"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/tool/tbot/config"
+	"github.com/gravitational/teleport/tool/tbot/identity"
+	"github.com/gravitational/trace"
+	"github.com/stretchr/testify/require"
+)
+
+// mockRunner is a mock tsh runner
+type mockRunner struct {
+	captureErr error
+	captureOut []byte
+}
+
+func (r *mockRunner) Capture(args ...string) ([]byte, error) {
+	if r.captureErr != nil {
+		return nil, r.captureErr
+	}
+	return r.captureOut, nil
+}
+
+func (r *mockRunner) Exec(env map[string]string, args ...string) error {
+	return trace.NotImplemented("not implemented")
+}
+
+// TestTSHSupported ensures that the tsh version check works as expected (and,
+// implicitly, that the version capture and parsing works.)
+func TestTSHSupported(t *testing.T) {
+	version := func(v string) []byte {
+		bytes, err := json.Marshal(struct {
+			Version string `json:"version"`
+		}{
+			Version: v,
+		})
+		require.NoError(t, err)
+
+		return bytes
+	}
+
+	tests := []struct {
+		name   string
+		out    []byte
+		err    error
```

Comment:
```suggestion
		mock *mockRunner
```

instead of dismembering the runner? Makes it easier to reason about what happens with out/err.